redis_collections/lists.py

```diff
@@ -601,3 +637,270 @@ def sync_trans(pipe):
             self._sync_helper(pipe)
 
         self._transaction(sync_trans)
+
+
+class Deque(List):
+    """A Redis-backed version of Python's :class:`collections.deque`.
+    See Python's `deque documentation
+    <https://docs.python.org/3/library/collections.html#collections.deque>`_
+    for more details. The Redis implementation is based on the `list type
+    <http://redis.io/commands#list>`_.
+    """
+
+    python_cls = collections.deque
+
+    def __init__(self, *args, **kwargs):
+        """
+        :param data: Initial data.
+        :type data: iterable
+        :param maxlen: Maximum size.
+        :type maxlen: int
+        :param redis: Redis client instance. If not provided, default Redis
+                      connection is used.
+        :type redis: :class:`redis.StrictRedis`
+        :param key: Redis key of the collection. Collections with the same key
+                    point to the same data. If not provided, default random
+                    string is generated.
+        :type key: str
+        :param writeback: If ``True`` keep a local cache of changes for storing
+                          modifications to mutable values. Changes will be
+                          written to Redis after calling the ``sync`` method.
+        :type key: bool
+
+        .. warning::
+            The ``maxlen`` of the collection can't be enforced when multiple
+            processes are accessing its Redis collection.
+        """
+        if len(args) > 2:
+            msg = '{} takes at most 2 positional arguments ({} given)'
+            raise TypeError(msg.format(self.__class__, len(args)))
+        elif len(args) == 2:
+            maxlen = args[1]
+            args = args[:1]
+        else:
+            maxlen = None
+
+        if (maxlen is not None) and not isinstance(maxlen, six.integer_types):
+            raise TypeError('an integer is required')
+
+        if (maxlen is not None) and maxlen < 0:
+            raise ValueError('maxlen must be non-negative')
+
+        self.maxlen = maxlen
+        super(Deque, self).__init__(*args, **kwargs)
+
+    # Magic methods
+
+    def __delitem__(self, index):
+        if isinstance(index, slice):
+            raise TypeError
+
+        return super(Deque, self).__delitem__(index)
+
+    def __getitem__(self, index):
+        if isinstance(index, slice):
+            raise TypeError
+
+        return super(Deque, self).__getitem__(index)
+
+    def __setitem__(self, index, value):
+        if isinstance(index, slice):
+            raise TypeError
+
+        return super(Deque, self).__setitem__(index, value)
+
+    # Named methods
+
+    def _append_helper(self, value, pipe):
+        # Append on the right
+        len_self = pipe.rpush(self.key, self._pickle(value))
```

Comment:
Could have re-used `List`'s `append` and `pop`, but that involved needless pickling.